perflib/perflib.go

```diff
@@ -231,6 +233,7 @@ func queryRawData(query string) ([]byte, error) {
 			newBuffer := make([]byte, len(buffer)+16384)
 			copy(newBuffer, buffer)
 			buffer = newBuffer
+			syscall.RegCloseKey(syscall.HKEY_PERFORMANCE_DATA)
```

Comment:
Nice catch, thanks! Why a separate close here, wouldn't the deferred call close any open keys when the function returns?

Reply:
Because the will be another RegQueryValueEx call right after this because of the continue statement. Also we tested without this line and it leaks.